runtime/bundles/org.eclipse.e4.core.di.extensions.supplier/src/org/eclipse/e4/core/di/internal/extensions/PreferencesObjectSupplier.java

```diff
@@ -177,9 +168,18 @@ private String getNodePath(IObjectDescriptor descriptor, Class<?> requestingObje
 		return nodePath;
 	}
 
-	private void addListener(String nodePath, String key, final IRequestor requestor) {
+	private Preferences getScope(Preference qualifier) {
+		String scope = qualifier.scope();
+		if (scope == null || scope.isEmpty()) {
+			scope = System.getProperty("DEFAULT_PREFERENCE_SCOPE", InstanceScope.INSTANCE.getName()); //$NON-NLS-1$
+		}
+		return getPreferencesService().getRootNode().node(scope);
+	}
+
+	private void addListener(Preferences scope, String nodePath, String key, final IRequestor requestor) {
 		if (requestor == null)
 			return;
+		String cacheKey = scope.name() + "/" + nodePath; //$NON-NLS-1$
```

Comment:
Some changes are missed here. This is the diff in dsg-4_11
![image](https://github.com/Halliburton-Landmark/eclipse.platform/assets/43205505/cf265406-b68e-4749-b504-03e52280a3d8)